### PR TITLE
Define QueryStatus and connect queries to BookmarkLoadResult

### PR DESCRIPTION
**Pull request by DavidS-om**

### CODE REVIEW

**Review by dylanratcliffe — APPROVED** [review]
I like this implementation, will need tickets created to mkae sure the TS and Go libraries both support it but this will help with progress tracking in the UI too

File: gateway.proto [hunk 1]
```diff
@@ -140,6 +142,9 @@ message LoadBookmark {
 message BookmarkLoadResult {
     bool success = 1;
     string errorMessage = 2;
+
+    // UUIDs of all queries that have been started as a result of loading this bookmark
+    repeated bytes startedQueries = 3;
```
Comment (dylanratcliffe) [inline gateway.proto:147]: Whoops missed this. I don't like the idea of multi-layering the serialisation, I think this should be just a `repeated Query`

**Review by dylanratcliffe — CHANGES_REQUESTED** [review]
Changed my ind!

File: gateway.proto (hunk 1 above)
Comment (DavidS-om) [reply gateway.proto:147]: That's only UUIDs, not full queries. Maybe:
```suggestion
    repeated bytes startedQueryUUIDs = 3;
```
?

Comment (dylanratcliffe) [reply gateway.proto:147]: Ohhhhh Yeah let's do that

**Review by dylanratcliffe — APPROVED** [review]

**Comment by DavidS-om** [comment]
> I like this implementation, will need tickets created to mkae sure the TS and Go libraries both support it but this will help with progress tracking in the UI too

https://github.com/overmindtech/sdp-js/pull/88 has the sdp-js bits. for sdp-go, I'll mature the WS client a bit in api-server before publishing it here, unless you had any other requirements?

**Pull request merged**